src/senselab/text/tasks/sentiment_analysis/sentiment_analysis.py

```diff
@@ -0,0 +1,91 @@
+"""Sentiment analysis implementation using the BaseAnalysis class."""
+
+from collections import defaultdict
+from typing import Any, Dict, List, Optional, Union
+
+from senselab.text.tasks.sentiment_analysis.constants import Sentiment
+from senselab.utils.data_structures.device import DeviceType, _select_device_and_dtype
+from senselab.utils.interfaces.analyses import BaseTextAnalysis
+from senselab.utils.model_utils import BaseModelSourceUtils
+from senselab.utils.tasks.chunking import chunk_text
+
+
+class SentimentAnalysis(BaseTextAnalysis):
+    """A class for performing sentiment analysis on text."""
+
+    @classmethod
+    def analyze(
+        cls,
+        input_data: List[Any],
+        model_utils: BaseModelSourceUtils,
+        device: Optional[DeviceType],
+        **kwargs: Any,  # noqa: ANN401
+    ) -> List[Dict[str, Any]]:
+        """Perform sentiment analysis on a list of text pieces.
+
+        Args:
+            input_data (List[Any]): List of text strings to analyze.
+            model_utils (BaseModelSourceUtils): Utility class for model operations.
+            device (Optional[DeviceType]): The device to use for computation (e.g., CPU, CUDA).
+            **kwargs (Any): Additional keyword arguments, such as:
+                - max_length (int): Maximum length of text chunks (default: 512).
+                - overlap (int): Overlap size between text chunks (default: 128).
+                - neutral_threshold (float): Threshold for considering sentiment as neutral (default: 0.05).
+
+        Returns:
+            List[Dict[str, Any]]: A list of dictionaries, each containing:
+                - score (float): Sentiment score between -1 and 1.
+                - label (str): The sentiment label ("negative", "neutral", "positive").
+
+        Raises:
+            ValueError: If the input list is empty or None.
+        """
+        if not input_data:
+            raise ValueError("Input list is empty or None.")
+
+        max_length = int(kwargs.get("max_length", 512))
+        overlap = int(kwargs.get("overlap", 128))
+        neutral_threshold = float(kwargs.get("neutral_threshold", 0.05))
+
+        device, torch_dtype = _select_device_and_dtype(
+            user_preference=device, compatible_devices=[DeviceType.CUDA, DeviceType.CPU]
+        )
+
+        tokenizer = model_utils.get_tokenizer(task="sentiment-analysis")
+        pipe = model_utils.get_pipeline(task="sentiment-analysis", device=device, torch_dtype=torch_dtype, top_k=None)

```

Comment:
is there an advantage in handling SentimentAnalysis with a different pipeline than "text-classification"?

Reply:
not really an advantage, it actually uses the same `TextClassificationPipeline` as `text-classification`. but I’ve noticed in the docs that `sentiment-analysis` is often mentioned as the task when building any sentiment analysis classifier. I guess that’s just to explicitly define the type of text classification. functionality-wise, it’s not really different.